images/virtualization-artifact/pkg/controller/moduleconfig/util.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package moduleconfig
+
+import (
+	"fmt"
+	"net/netip"
+
+	corev1 "k8s.io/api/core/v1"
+
+	mcapi "github.com/deckhouse/virtualization-controller/pkg/controller/moduleconfig/api"
+)
+
+func isEqualCIDRs(a, b netip.Prefix) bool {
+	return a.Addr() == b.Addr() && a.Bits() == b.Bits()
+}
+
+func checkOverlapsCIDRs(prefixes []netip.Prefix) error {
+	for i := 0; i < len(prefixes); i++ {
+		for j := i + 1; j < len(prefixes); j++ {
+			if prefixes[i].Overlaps(prefixes[j]) {
+				return fmt.Errorf("overlapping CIDRs %v and %v", prefixes[i], prefixes[j])
+			}
+		}
+	}
+	return nil
+}
+
+func checkNodeAddressesOverlap(nodes []corev1.Node, excludedPrefixes []netip.Prefix) error {
+	for _, node := range nodes {
+		for _, address := range node.Status.Addresses {
+			if address.Type == corev1.NodeInternalIP || address.Type == corev1.NodeExternalIP {
+				ip, err := netip.ParseAddr(address.Address)
+				if err != nil {
+					return fmt.Errorf("error parsing node IP: %w", err)
```

Comment:
It should not be the "stop everything" error, this validator checks if nodes IPs are overlapped with CIDRs in ModuleConfig. If IP is not parsed it is ok.